### PR TITLE
Make API key required in config

### PR DESCRIPTION
**Pull request by davidkpiano**
To simplify the API and make it possible to run multiple machines from different projects, we shouldn't implicitly grab the API key from `.env`, but instead do what other APIs do, which is require them directly.

We should direct users to put these API keys in their environment variables file and then read it in via `process.env.ANY_API_KEY_VAR`.

### CODE REVIEW

File: packages/sky-core/src/utils.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
 import PartySocket from 'partysocket';
 import superjson from 'superjson';
-import { SKY_API_URL, SKY_HOST } from './env';
 import { SkyClientEvent } from './types';
 
 export const sendToSky = (socket: PartySocket, event: SkyClientEvent) => {
   socket.send(superjson.stringify(event));
 };
 
+let SKY_HOST = 'stately-sky-beta.mellson.partykit.dev';
```
Comment (gavination) [inline packages/sky-core/src/utils.ts:9]: Don't think this hardcoded value should be here, no?

**Review by gavination — CHANGES_REQUESTED** [review]

**Review by mellson — CHANGES_REQUESTED** [review]
I think this is a good change. Can you:

1. Also update the [`env.ts`](https://github.com/statelyai/sky/blob/main/packages/sky-core/src/env.ts) file to reflect the change
2. Revert the `yarn-error` & `pnpm-lock` files.
3. We also need to update [the documentation](https://stately.ai/docs/stately-sky-getting-started) to reflect this, would you like me to handle that part?

**Comment by mellson** [comment]
> ... and make it possible to run multiple machines from different projects
We need to update the `xstate sky` cmd for this to work properly. I've created [STA-5554](https://linear.app/statelyai/issue/STA-5554) for this work.

File: packages/sky-core/src/utils.ts (hunk 1 above)
Comment (mellson) [reply packages/sky-core/src/utils.ts:9]: It's fine, it should be there 👍🏻

File: packages/sky-core/src/actorFromStately.ts [hunk 2]
```diff
@@ -39,7 +38,6 @@ export async function actorFromStately<T extends AnyStateMachine>(
   // Get the host for Partykit and the API base URL for Stately Studio
   const { host, apiBaseURL } = skyConnectionInfo();
   const { actorId, machine } = skyConfig;
-  const apiKey = _apiKey ?? SKY_API_KEY;
 
   if (!apiKey) {
```
Comment (mellson) [inline packages/sky-core/src/actorFromStately.ts:42]: This if block can now be removed

File: vite-env.d.ts [hunk 3]
```diff
@@ -6,7 +6,5 @@ interface ImportMeta {
     PROD: boolean;
     DEV: boolean;
     VITE_SKY_HOST?: string;
-    VITE_SKY_API_URL?: string;
```
Comment (mellson) [inline vite-env.d.ts:9]: We want to keep `VITE_SKY_API_URL`

**Comment by mellson** [comment]
@davidkpiano Instead of you having to handle all our comments here I opened a new PR with this change here #14

**Pull request closed without merging**